lib/src/widgets/country_selector/search_box.dart

```diff
@@ -18,21 +18,47 @@ class SearchBox extends StatelessWidget {
     this.searchIconColor,
   }) : super(key: key);
 
+  @override
+  State<SearchBox> createState() => _SearchBoxState();
+}
+
+class _SearchBoxState extends State<SearchBox> {
+  String _previousValue = '';
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
   @override
   Widget build(BuildContext context) {
     return Padding(
       padding: const EdgeInsets.fromLTRB(16, 16, 16, 0),
       child: TextField(
-        autofocus: autofocus,
-        onChanged: onChanged,
-        onSubmitted: (_) => onSubmitted(),
-        cursorColor: style?.color,
-        style: style,
-        decoration: decoration ??
+        autofocus: widget.autofocus,
+        onChanged: (e) {
+          widget.onChanged(e);
+
+          // detect length difference
+          final diff = e.length - _previousValue.length;
+          if (diff > 3) {
+            // more than 3 characters added, probably a paste / autofill of country name
```

Comment:
Is this guaranteed ?

Reply:
We can adjust the threshold if you'd like. I have tested adding fast and I did not experience any issues, probably given the tick-rate of the listener. However, if you feel more comfortable increasing this to e.g. 5 or 6 letters, that should be good to for most cases.

We do have quite a number of countries with only 4 letters though

Reply:
yeah 3 seems fine, however, I believe the current way this works will select the first element in the list, doesn't it ?

If that's the case there might be issues when typing "Aus" then selecting Austria in the auto fill, it might select the first one which would be Australia

Reply:
Actually if you paste in "aus" nothing will happen, because it's not greater than 3. Also I guess that would also only be an edge-case. Who would copy paste "aust" in? If you were already pasting, chances are you would paste in "Australia" or "Austria". And autofill certainly would not auto-paste "aust"

Reply:
That makes sense